src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/PseudoRoleToAccessProfileGenerator.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.RoleToAccessProfileDefinition;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.data.caseaccess.GlobalCaseRole.CREATOR;
+
+@Component
+public class PseudoRoleToAccessProfileGenerator {
+
+    protected static final String IDAM_PREFIX = "idam:";
+    private static final String CASE_ROLE_ID_REGEX = "^(\\[[A-Za-z]+\\])$";
```

Comment:
Is this correct? Can't recall if I've seen some services using case roles with numbers in them. Should verify what is currently accepted by the Definition Store.

Reply:
Checked and this was taken directly from def store 👍